projects/ccd-case-ui-toolkit/src/lib/shared/components/case-viewer/services/case.resolver.ts

```diff
@@ -85,22 +85,27 @@ export class CaseResolver implements Resolve<CaseView> {
           this.caseNotifier.announceCase(this.caseNotifier.cachedCaseView);
           return this.caseNotifier.cachedCaseView;
         }),
-        catchError(error => this.checkAuthorizationError(error))
+        catchError(error => this.checkAuthorizationError(error, cid))
       ).toPromise();
   }
 
-  private checkAuthorizationError(error: any) {
+  private checkAuthorizationError(error: any, caseReference: string) {
+    console.error(error);
     // TODO Should be logged to remote logging infrastructure
     if (error.status === 400) {
       this.router.navigate(['/search/noresults']);
       return of(null);
     }
-    console.error(error);
+    // Error 403, navigate to restricted case access page
```

Comment:
I think I would move this to after the CaseResolver.EVENT_REGEX block below. I don't imagine that an event trigger would return 403 very often, but if it did then you'd trigger the restricted case logic, which may not be what you want

Reply:
Thank you @andywilkinshmcts, I have made the changes as you suggested.